### ISSUE TITLE
Prefix option not honored in client compilation & installation

### ISSUE THREAD

**Issue by mdatre** [BOINC/boinc#3965]
**Describe the bug**
Make install tries to put client service file in the root space, despite giving --prefix=/different/local/path/than/root to the configure script while compiling. IMO this is not the expected behaviour. With --prefix option, the compiled client binaries should be put in the /different/local/path/than/root folder, and it let user start the client manually, than as a system service by putting it in the root space.

**Steps To Reproduce**
1. Git clone boinc repo 
2. $ cd /path/to/boinc/repo
3. $ ./_autosetup 
4. $ ./configure --prefix=/different/local/path
5. $ make install
`/usr/bin/install: cannot create regular file '/usr/lib/systemd/system/boinc-client.service': Permission denied
make[4]: *** [Makefile:562: install-exec-hook] Error 1
make[4]: Leaving directory '/work/boinc/client/scripts'
make[3]: *** [Makefile:503: install-exec-am] Error 2
make[3]: Leaving directory '/work/boinc/client/scripts'
make[2]: *** [Makefile:453: install-am] Error 2
make[2]: Leaving directory '/work/boinc/client/scripts'
make[1]: *** [Makefile:649: install-recursive] Error 1
make[1]: Leaving directory '/work/boinc'
make: *** [Makefile:951: install] Error 2
`

**Expected behavior**
Should compile and install any binaries under /different/local/path when given --prefix option to configure script.

**System Information**
 - OS: Linux Ubuntu 20.04
 - BOINC Version: 7.17.0

**Additional context**
With congure using --prefix option, make install should not try to put things in root space like /usr/lib/...

/usr/bin/install: cannot create regular file '/usr/lib/systemd/system/boinc-client.service': Permission denied
make[4]: *** [Makefile:562: install-exec-hook] Error 1
make[4]: Leaving directory '/work/boinc/client/scripts'
make[3]: *** [Makefile:503: install-exec-am] Error 2
make[3]: Leaving directory '/work/boinc/client/scripts'
make[2]: *** [Makefile:453: install-am] Error 2
make[2]: Leaving directory '/work/boinc/client/scripts'
make[1]: *** [Makefile:649: install-recursive] Error 1
make[1]: Leaving directory '/work/boinc'
make: *** [Makefile:951: install] Error 2

**Comment by RichardHaselgrove** [comment]
> Permission denied

What are the ownership and write permissions on your systemd directory tree? What user rights did the operator of 'make install' have?

This sounds like an operating system problem, not a BOINC problem.

**Comment by mdatre** [comment]
Boinc is being compiled and installed as a local user, without using sudo/root permissions. That user does not have (or does not want to use) the write permissions on the systemd directory. That is why the user has given --prefix=/different/local/path option to the configure script in Boinc sources. Yet "make install" tries to write boinc-client.service to the /usr/lib/systemd/ folder.

I do not think this is an OS issue. Doesn't a normal user have an option to do just a **local** install without disturbing or installing things in the '/' space? (That's what --prefix was for, I assume).

**Comment by RichardHaselgrove** [comment]
The whole systemd installation procedure was added in PR #2260, with significant input from Linux maintainers @Germano0 and @LocutusOfBorg. Looking specifically at https://github.com/BOINC/boinc/pull/2260#issuecomment-361889905, one motivation was to ensure 'confinement' of BOINC according to SELinux principles.

At the time, I was acting as Release Manager for the whole release cycle (all platforms), so I participated in the thread: but I am not a Linux specialist, and at the time I didn't even have access to a Linux machine for testing.

There is diversity of opinion within the BOINC community on this issue. Some voices come from a community which views the Linux client as a tool in a controlled, corporate, environment - hence, for example, a corporate end-user may not have the authority to re-start a system service after stopping it, and the option to "Shut down connected client" has been removed from some distributions (though not from BOINC's own source code).

Other voices come from a community of individuals and home-based users who **do** have full sysadmin control over their own machines, and are content to allow BOINC to run in an unconfined mode. It seems that this issue touches - yet again - on the differences between these views. Perhaps this is less a 'bug', but more a design decision which has veered closer to one view than the other.

**Comment by AenBleidd** [comment]
@RichardHaselgrove, I'm not sure you're completely right. As far as I know, using `systemd` doesn't mean that you can use application in a classic way. So this ticket is smth that could be fixed.

**Comment by RichardHaselgrove** [comment]
I wasn't really expressing a view on the 'fixability' of the reported problem, just supplying some background about how it might have arisen, and hence where we might need to look to fix it.

I do now have two Linux machines, and I have compiled my own client without encountering this problem. But I ran 'make', not 'make install' - so my binary file ended up in the user area for final disposition under manual control. The systemd assumption and placement seems to come from the 'install' operation, not the 'make' operation.

**Comment by mdatre** [comment]
That's right, but if I run configure and then make install by disabling client compilation as:

`$ ./configure --disable-client --disable-manager --prefix=/different/local/path`
`$ make install`

It works by putting any compiled libraries and binaries in `/different/local/path`.

So should it not be the same behaviour for client compilation and install as well?

Or is the user supposed to **not** run "make install" at all (with or without client compilation), and is expected to move the generated binaries manually from the Boinc source folder to whichever location of user's choice?

**Comment by RichardHaselgrove** [comment]
I'd suggest that 'user compilation' wasn't considered in the original PR at all: the contributor's mind was focussed on 'repo compilation/distribution'.

So, if there's a difference between the --disable-client pathway and full compilation pathways, that is indeed a bug and should be fixed.

**Comment by mdatre** [comment]
Hi, just checking if this is going to be fixed anything soon?

**Comment by truboxl** [comment]
Use `make install DESTDIR=/different/local/path` as a workaround
